addons/sale_loyalty/models/sale_order.py

```diff
@@ -913,8 +913,16 @@ def _program_check_compute_points(self, programs):
                         points += rule.reward_point_amount
                     elif rule.reward_point_mode == 'money':
                         # Compute amount paid for rule
-                        # NOTE: this does not account for discounts -> 1 point per $ * (100$ - 30%) will result in 100 points
-                        amount_paid = sum(max(0, line.price_total) for line in order_lines if line.product_id in rule_products)
+                        # NOTE: this accounts for discounts -> 1 point per $ * (100$ - 30%) will result in 70 points
+                        lines_for_points = []
+                        for line in self.order_line - self._get_no_effect_on_threshold_lines():
+                            is_discount = line.reward_id.reward_type == 'discount'
+                            reward_program = line.reward_id.program_id
+                            if line.reward_id and reward_program.program_type in ['ewallet', 'gift_card', program.program_type]:
+                                continue
+                            lines_for_points.append(line)
+
+                        amount_paid = sum(line.price_total for line in lines_for_points if line.product_id in so_products_per_rule.get(rule, []))
```

Comment:
Since we already loop over the lines, let's sum them directly :)
```suggestion
                        # NOTE: this accounts for discounts -> 1 point per $ * (100$ - 30%) will result in 70 points
                        lines_for_points = []
                        amount_paid = 0.0
                        rule = so_products_per_rule.get(rule, [])
                        for line in self.order_line - self._get_no_effect_on_threshold_lines():
                            is_discount = line.reward_id.reward_type == 'discount'
                            reward_program = line.reward_id.program_id
                            if line.reward_id and reward_program.program_type in ['ewallet', 'gift_card', program.program_type]:
                                continue
                            lines_for_points.append(line)
                            amount_paid += line.price_total if line.product_id in rule else 0.0
```

Reply:
Thank you ! I've changed the code locally to include this suggestion. I'll push it once I also include changes based on Itinel's review